### PR TITLE
Filtering out invalid regions

### PR DESCRIPTION
**Pull request by spfink**
Filters out any regions that don't match the partition regex or don't match the partition regex with "-fips" at the end, or don't have global in the name.

## Testing
Updated existing tests to reflect removed regions.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the **CONTRIBUTING** document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [x] My change requires a change to the Javadoc documentation
- [x] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] A short description of the change has been added to the **CHANGELOG**
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: codegen-lite/src/main/java/software/amazon/awssdk/codegen/lite/regions/ServiceMetadataGenerator.java [hunk 1]
```diff
@@ -214,16 +221,34 @@ private MethodSpec signingRegion() {
...
+     *
+     * @param regex - Regex for regions in a given partition.
+     * @param region - Region string being checked.
+     * @return true if the region string should be included as a region.
+     */
+    private boolean validRegion(String regex, String region) {
+        return region.matches(regex) ||
+               region.endsWith("-fips") &&
+               region.matches(regex.replace("$", "(-fips)*$")) ||
+               region.contains("global");
```
Comment (millems) [inline codegen-lite/src/main/java/software/amazon/awssdk/codegen/lite/regions/ServiceMetadataGenerator.java]: Can we add parens, methods or variables to make it more obvious what the precedence is here?

File: codegen-lite/src/main/java/software/amazon/awssdk/codegen/lite/regions/ServiceMetadataGenerator.java [hunk 2]
```diff
@@ -214,16 +221,34 @@ private MethodSpec signingRegion() {
...
+     * and match the partition regex with that included, or include the word "global".
+     *
+     * @param regex - Regex for regions in a given partition.
+     * @param region - Region string being checked.
+     * @return true if the region string should be included as a region.
+     */
+    private boolean validRegion(String regex, String region) {
+        return region.matches(regex) ||
+               region.endsWith("-fips") &&
+               region.matches(regex.replace("$", "(-fips)*$")) ||
```
Comment (millems) [inline codegen-lite/src/main/java/software/amazon/awssdk/codegen/lite/regions/ServiceMetadataGenerator.java]: This line seems more complex than it needs to be. Why not just substring off the -fips in the region?

Comment (millems) [inline codegen-lite/src/main/java/software/amazon/awssdk/codegen/lite/regions/ServiceMetadataGenerator.java]: Can we write some unit tests for this specific logic?

Comment (millems) [inline codegen-lite/src/main/java/software/amazon/awssdk/codegen/lite/regions/ServiceMetadataGenerator.java]: Have we verified how this affects existing regions?

Comment (spfink) [reply codegen-lite/src/main/java/software/amazon/awssdk/codegen/lite/regions/ServiceMetadataGenerator.java]: I've verified with the services that I know had issues that this works. 

It does however introduce some backwards incompatibilies.

Regions like sandbox and local are gone. Those are nice to have but we'd have to one off them.

I don't want to but we may want to have boths -fips as a suffix and fips- as a prefix. S3 has fips- as a suffix and we are removing those...

File: codegen-lite/src/main/java/software/amazon/awssdk/codegen/lite/regions/ServiceMetadataGenerator.java (hunk 1 above)
Comment (spfink) [reply codegen-lite/src/main/java/software/amazon/awssdk/codegen/lite/regions/ServiceMetadataGenerator.java]: Refactored out to a utility class for clarity and testability.

File: codegen-lite/src/main/java/software/amazon/awssdk/codegen/lite/regions/ServiceMetadataGenerator.java (hunk 2 above)
Comment (spfink) [reply codegen-lite/src/main/java/software/amazon/awssdk/codegen/lite/regions/ServiceMetadataGenerator.java]: Refactored out to a utility class for clarity and testability.

Comment (spfink) [reply codegen-lite/src/main/java/software/amazon/awssdk/codegen/lite/regions/ServiceMetadataGenerator.java]: Refactored out to a utility class for clarity and testability.

**Pull request merged**